### PR TITLE
Rent model

### PR DESCRIPTION
**Pull request by erosennin93**
## Review

Add Rent model and its migrations.

# Trello Card
https://trello.com/c/cxAVAhCc/19-crear-modelo-de-rent

# Screenshots

 - Create data
image

 - Get User rents
image (1)

### CODE REVIEW

**Comment by erosennin93** [comment]
@leamotta the original base from this branch was _branch-model_ , I just merge and resolved the conflict

File: app/models/rent.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class Rent < ApplicationRecord
+  validates :begin, :end, presence: true
```
Comment (leamotta) [inline app/models/rent.rb]: we should validate that the user and book are present as well

File: db/migrate/20191219132204_create_rents.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class CreateRents < ActiveRecord::Migration[6.0]
+  def change
+    create_table :rents do |t|
+      t.references :user, foreign_key: true
+      t.references :book, foreign_key: true
```
Comment (leamotta) [inline db/migrate/20191219132204_create_rents.rb]: validate at DB level as well with `null: false`

File: app/models/rent.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class Rent < ApplicationRecord
+  validates :user, :book, :begin, :end, presence: true
```
Comment (leamotta) [inline app/models/rent.rb]: the card does not ask for it but with your knowledge so far you should be able to add a custom validation saying that the `begin`  date should be before the `end` date
When you are modeling stuff you should always think about this kind of logic

**Comment by leamotta** [comment]
just that and we're good!

**Pull request merged**